src/Tasks/PageIterator.php

```diff
@@ -109,34 +112,36 @@ public function next(): ?PageResult {
     /**
      * @param $response
      * @return PageResult|null
+     * @throws JsonException
      */
     public static function convertToPage($response): ?PageResult {
         $page = new PageResult();
         if ($response === null) {
             throw new InvalidArgumentException('$response cannot be null');
         }
 
-        $value = is_array($response) ? $response['value'] : $response->value;
+        if (is_array($response)) {
+            $value = $response['value'];
+        } else if(is_a($response, Parsable::class) && method_exists($response, 'getValue')) {
+            $value = $response->getValue();
+        } else {
+            $value = $response->value;
+        }
 
         if ($value === null) {
-            throw new InvalidArgumentException('');
+            throw new InvalidArgumentException('The response does not contain a value.');
         }
 
-        $collected = [];
-
-        for ($i = 0; $i < count($value); $i++) {
-            $collected []= $value[$i];
-        }
-        $parsablePage =  is_a($response, Parsable::class) ? $response : json_decode(json_encode($response), true);
+        $parsablePage =  is_a($response, Parsable::class) ? $response : json_decode(json_encode($response,JSON_THROW_ON_ERROR), true);
         if (is_array($parsablePage)) {
             $page->setOdataNextLink($parsablePage['@odata.nextLink'] ?? '');
         } else {
             $page->setOdataNextLink($parsablePage->getOdataNextLink());
```

Comment:
Perhaps some safeguards here since we found that some collection response objects may not have the `OdataNextLink` property?